pkg/quota/image/imagestream_evaluator.go

```diff
@@ -31,13 +33,21 @@ func NewImageStreamEvaluator(isNamespacer osclient.ImageStreamsNamespacer) kquot
 		ConstraintsFunc:      generic.ObjectCountConstraintsFunc(imageapi.ResourceImageStreams),
 		UsageFunc:            generic.ObjectCountUsageFunc(imageapi.ResourceImageStreams),
 		ListFuncByNamespace: func(namespace string, options kapi.ListOptions) ([]runtime.Object, error) {
-			itemList, err := isNamespacer.ImageStreams(namespace).List(options)
+			list, err := store.ImageStreams(namespace).List(options.LabelSelector)
 			if err != nil {
 				return nil, err
 			}
-			results := make([]runtime.Object, 0, len(itemList.Items))
-			for i := range itemList.Items {
-				results = append(results, &itemList.Items[i])
+			results := make([]runtime.Object, 0, len(list))
+			for _, is := range list {
+				objCopy, err := kapi.Scheme.DeepCopy(is)
```

Comment:
> If I don't I might end up modifying cache, and that's not desired, right?

Quota counts won't do mutation.  In general, we've tried to avoid unnecessary deep copies, so I'm used to seeing a pattern in controllers where you check whether you are likely to have changes against your shared cache reference.  After you've eliminated all the obvious pre-checks, you can make a copy in one spot to make the code easier to reason about, but it should be minority of cases.

Doing this copy here would be too early for a normal controller and we happen to know in this case we don't need it, so I would skip the copies.